### ISSUE TITLE
Compile-Time Data is Being Syntax Colorized and Syntax Checked

### ISSUE THREAD

**Issue by BrianGodsend** [codefori/vscode-ibmi#2421]
The editor is colorizing and (at times) syntax checking compile-time data in RPGLE and SQLRPGLE source members.

For example:
```
**FREE
dcl-s srcDta           Char(100) dim(*CTDATA);
dcl-s i                Int(10:0);
for i = 1 to %elem(srcDta);
endfor;
*inLR = *ON;
return;
**CTDATA srcDta
      * Test's stuff
     d junk            ds                  qualified inz
     d  field                         1a
   // Other's stuff
 dcl-ds junk  qualified inz;
field        Char(1);
   end-ds;
              dcl-s somevar  Varchar(5000);
```
In the above example, the compile-time data for `srcDta` is colorized.  Additionally, note:

* The colorization is messed up by the presence of the single single-quote in `Test's stuff`.  It appears that despite not having any line continuation characters at the end of the line, the editor thinks the single-quote is the start of a multi-line literal.
* In this case, the indentation errors are not being thrown.  In other very lengthy and complex examples, I have seen the compile-time data not only be colorized, but errors (at least indentation errors) be flagged.

Compile-time data should not be colorized nor syntax checked.

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.14.5|
|Visual Studio Code version|1.96.0|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
.NET Install Tool (vscode-dotnet-runtime): 2.2.3
COBOL (cobol): 24.11.28
Code Coverage for IBM i (code-coverage-ibmi): 0.1.7
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.7.0
Dev Containers (remote-containers): 0.394.0
ESLint (vscode-eslint): 3.0.10
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.20.0
Extension Authoring (extension-editing): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
GitHub Pull Requests (vscode-pull-request-github): 0.102.0
GitLens — Git supercharged (gitlens): 16.0.5
HTML CSS Support (vscode-html-css): 2.0.11
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
IBM i Project Explorer (vscode-ibmi-projectexplorer): 2.12.0
JSON Language Features (json-language-features): 1.0.0
Jira and Bitbucket (Atlassian Labs) (atlascode): 3.0.15
Merge Conflict (merge-conflict): 1.0.0
NPM support for VS Code (npm): 1.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Nx Console (angular-console): 18.33.1
Path Intellisense (path-intellisense): 2.10.0
Prettier - Code formatter (prettier-vscode): 11.0.0
RPGLE (vscode-rpgle): 0.28.2
Source Orbit (vscode-sourceorbit): 1.0.2
TODO Highlight (vscode-todo-highlight): 1.0.5
Todo Tree (todo-tree): 0.0.226
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL (remote-wsl): 0.88.5
YAML (vscode-yaml): 1.15.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|7|
|CCSID Origin|37|
|Runtime CCSID|37|
|Default CCSID|37|
|SSHD CCSID|?|
|cqsh|true|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|
|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|startDebugService.sh|64bit|
|chsh|iconv|||||
|find|setccsid|||||
|git|uname|||||
|grep||||||
|ls||||||
|md5sum||||||
|sort||||||
|stat||||||
|tar||||||
|tn5250||||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=QGPL
CURLIB=QGPL
PASE_USRGRP_LIMITED=N
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=10.1.14.31 54968 22
SSH_CONNECTION=10.1.14.31 54968 10.1.1.22 22
TZ=<EST>5<EDT>,M3.2.0,M11.1.0
_=/QOpenSys/usr/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$",
  "qsysNameRegex": {}
}
```
</details>

**Comment by worksofliam** [comment]
This might be an issue for ibmi-languages and not vscode-rpgle.

**Comment by BrianGodsend** [comment]
It looks like a related issue was transfered to ibmi-languages by [sebjulliand](https://github.com/sebjulliand).

Is this something I can do, or should I leave it to the experts?

**Comment by chrjorgensen** [comment]
@BrianGodsend Your issue has been fixed in [vscode-ibmi-languages version 0.6.20](https://github.com/barrettotte/vscode-ibmi-languages/releases/tag/v0.6.20) - please upgrade.

After upgrade, compile time data will be shown like this:

![billede](https://github.com/user-attachments/assets/e41a2607-4b0d-4d19-9230-fe50709d4c71)
